nemo/collections/asr/parts/submodules/ctc_beam_decoding.py

```diff
@@ -416,6 +435,102 @@ def _pyctcdecode_beam_search(
+            )
+
+        if self.flashlight_beam_scorer is None:
+            # Check for filepath
+            if self.kenlm_path is None or not os.path.exists(self.kenlm_path):
+                raise FileNotFoundError(
+                    f"KenLM binary file not found at : {self.kenlm_path}. "
+                    f"Please set a valid path in the decoding config."
+                )
+
+            # perform token offset for subword models
+            #if self.decoding_type == 'subword':
+            #    vocab = [chr(idx + self.token_offset) for idx in range(len(self.vocab))]
+            #else:
+            #    # char models
+            #    vocab = self.vocab
+
+            # Must import at runtime to avoid circular dependency due to module level import.
+            from nemo.collections.asr.modules.flashlight_decoder import FlashLightKenLMBeamSearchDecoder
+
+            self.flashlight_beam_scorer = FlashLightKenLMBeamSearchDecoder(
+                lm_path=self.kenlm_path,
+                vocabulary=self.vocab,
+                tokenizer=self.tokenizer,
+                lexicon_path=self.flashlight.lexicon_path,
+                nbest=self.flashlight.nbest,
+                beam_size=self.beam_size,
+                beam_size_token=self.flashlight.beam_size_token,
+                beam_threshold=self.flashlight.beam_threshold,
+                lm_weight=self.beam_alpha,
+                word_score=self.beam_beta,
+                unk_weight=self.flashlight.unk_weight,
+                sil_weight=self.flashlight.sil_weight,
+                unit_lm=self.flashlight.unit_lm,
+            )
+
+        x = x.to('cpu')
+
+        with typecheck.disable_checks():
+            beams_batch = self.flashlight_beam_scorer.forward(log_probs=x)
+
+        # For each sample in the batch
+        nbest_hypotheses = []
+        for beams_idx, beams in enumerate(beams_batch):
+            # For each beam candidate / hypothesis in each sample
+            hypotheses = []
+            for candidate_idx, candidate in enumerate(beams):
+                hypothesis = rnnt_utils.Hypothesis(
+                    score=0.0, y_sequence=[], dec_state=None, timestep=[], last_token=None
+                )
+
+                # For subword encoding, NeMo will double encode the subword (multiple tokens) into a
+                # singular unicode id. In doing so, we preserve the semantic of the unicode token, and
+                # compress the size of the final KenLM ARPA / Binary file.
+                # In order to do double encoding, we shift the subword by some token offset.
+                # This step is ignored for character based models.
+                #if self.decoding_type == 'subword':
+                #    pred_token_ids = [ord(c) - self.token_offset for c in candidate[1]]
+                #else:
+                #    pred_token_ids = candidate[1]
```

Comment:
Probably the commented stuff can be removed safely, we have the default code for reference if needed in the future

Reply:
Done